### PR TITLE
JAX docs tutorial: JAX internals 301 - JAX primitives

### PR DESCRIPTION
**Pull request by 8bitmp3**
@jakevdp 

https://github.com/google/jax/issues/18585

### CODE REVIEW

File: docs/tutorials/jax-primitives.md [hunk 1]
```diff
@@ -0,0 +1,592 @@
...
+kernelspec:
+  display_name: Python 3
+  language: python
+  name: python3
+---
+
+(jax-internals-jax-primitives)=
+# JAX internals 301: JAX primitives
+
+This tutorial explains the interface that a JAX primitive must support to allow JAX to perform all its transformations. JAX implements certain composable transformations of Python functions, such as {func}`jax.jit`, {func}`jax.grad` and {func}`jax.vmap`, in a *JAX-traceable* way. This means that when a Python function is executed, the only operations it applies to the data are either:
```
Comment (jakevdp) [inline docs/tutorials/jax-primitives.md:18]: Have we defined anywhere what a "primitive" is? If not, we should do that here. If so, we should link to it.

Comment (8bitmp3) [reply docs/tutorials/jax-primitives.md:18]: Will rephrase the original, since it was not 100% clear.

Comment (8bitmp3) [reply docs/tutorials/jax-primitives.md:18]: :+1: :+1: Thought about this too

**Review by jakevdp — APPROVED** [review]

**Comment by jakevdp** [comment]
We're getting some warnings in the doc build:
```
/home/runner/work/jax/jax/docs/tutorials/index.rst:52: WARNING: toctree contains reference to nonexisting document 'tutorials/custom-jaxprs-interpreters'
/home/runner/work/jax/jax/docs/tutorials/index.rst:52: WARNING: toctree contains reference to nonexisting document 'tutorials/type-promotion'
looking for now-outdated files... none found
pickling environment... done
/home/runner/work/jax/jax/docs/tutorials/jax-internals.md: WARNING: document isn't included in any toctree
```

**Pull request merged**